### ISSUE TITLE
UI state persistence

### ISSUE THREAD

**Issue by kasemir** [shroffk/phoebus#15]
When applications stop, they need some way to persist information,
and when applications start, they need some way to retrieve it.

The location & size of each stage as well as the current set of tabs should be persisted the same way on shutdown, then automatically starting those applications on restart.

This likely means every application instance needs a unique ID to associate it with its persisted information.
Do apps need a way to opt out, so they're not restored after phoebus restart?

**Comment by kasemir** [comment]
The location of all windows, the tabs that were open, the input file for tabs that do have an input are persisted.

By making the .phoebus/memento read-only, or by always putting a "golden" file in place before starting phoebus one can get the behavior where it always starts in a known layout.

"Perspectives" would be a separate issue, but the basic mechanism for saving and then restoring a layout are there, just for now using one setting automatically, instead of switching between multiple settings at runtime.
